src/main/groovy/org/kordamp/gradle/plugin/jandex/tasks/JandexWorkAction.groovy

```diff
@@ -97,7 +101,11 @@ abstract class JandexWorkAction implements WorkAction<JandexWorkParameters> {
             throw e
         }
 
-        private void indexFile(File file) throws IOException {
+        void indexFiles(Indexer indexer) {
+            classFiles.values().forEach { indexFile(indexer, it) }
```

Comment:
Given that this is the actual change required for creating a reproducible index, I don't see a strong reason against keeping the Indexer as a constructor argument. In doing so changes would be minimal compared to the current solution.

Reply:
Fair point, updated the PR